### PR TITLE
fix(client-src): don't use self.location.port

### LINKED ISSUE #1777: V3.3.0 live reload broken when set hostname to `0.0.0.0:xxxx`
- Operating System: MacOS
- Node Version: v10.15.0
- NPM Version: 6.4.`
- webpack Version: 4.29.6
- webpack-dev-server Version: 3.3.0

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

https://github.com/webpack/webpack-dev-server/blob/99e8db0f8cc441bb8f6acc60d786e33c38b8fc1d/client-src/default/index.js#L210

When start a node server like 0.0.0.0:8080, and load resources from `0.0.0.0:9000`, Port was reset to `self.location.port `, which actually is 8080.

### For Bugs; How can we reproduce the behavior?

- Start webpack-dev-server and listen at e.g. 0.0.0.0:9000
- Load  resources from another local server like: 0.0.0.0:8080

### Expected Behavior

Live reload should listen 0.0.0.0:9000

### Actual Behavior

Live reload listen 0.0.0.0:8080

### PR DESCRIPTION
**Pull request by hiroppy**
ISSUE: https://github.com/webpack/webpack-dev-server/issues/1777
REF: https://github.com/webpack/webpack-dev-server/pull/1792
REF: https://github.com/webpack/webpack-dev-server/pull/1664

- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Revert from #1664.

### Motivation / Use-Case

this pr is just a revert, but it is WIP status.(to validate)

### Breaking Changes

nope

### Additional Info

fixes #1777

### CODE REVIEW

**Comment by hiroppy** [comment]
I checked using @njugray's [repository](https://github.com/njugray/webpack-live-reload) and confirmed that this problem has been fixed.

**Comment by hiroppy** [comment]
hmmm, since my tests are passing locally, I will investigate this issue...

**Comment by hiroppy** [comment]
Yep, it is a difficult problem(because it just occurs on CI, and a port problem) so maybe I will use enough time.

**Comment by evilebottnawi** [comment]
@hiroppy friendly ping

File: test/Client.test.js [hunk 1]
```diff
@@ -62,7 +71,7 @@ describe('Client code', () => {
           .waitForRequest((requestObj) => requestObj.url().match(/sockjs-node/))
```
Comment (hiroppy) [inline test/Client.test.js]: When the port is `9000`, `requestObj.url()` is `http://localhost:9000/main.js` and then `requestObj.url()` will be `http://localhost:9001/sockjs-node/info?t=xxxx` after this page will be proxyed.

**Comment by hiroppy** [comment]
@evilebottnawi PTAL

**Comment by evilebottnawi** [comment]
Good job, thanks!

**Pull request merged**